app/controllers/navigation/Navigator.scala

```diff
@@ -412,10 +412,13 @@ object Navigator {
       controllers.declaration.routes.RepresentativeStatusController.displayPage(mode)
 
   private def carrierEoriNumberClearancePreviousPage(cacheModel: ExportsDeclaration, mode: Mode): Call =
-    if (cacheModel.parties.declarantIsExporter.exists(_.isExporter))
-      controllers.declaration.routes.DeclarantExporterController.displayPage(mode)
-    else {
+    if (!cacheModel.parties.declarantIsExporter.exists(_.isExporter))
       controllers.declaration.routes.RepresentativeStatusController.displayPage(mode)
+    else {
+      if (cacheModel.parties.consignorDetails.flatMap(_.details.address).isDefined)
+        controllers.declaration.routes.ConsignorDetailsController.displayPage(mode)
+      else
+        controllers.declaration.routes.ConsignorEoriNumberController.displayPage(mode)
```

Comment:
Why the previous page relates to Consignor now? Was there a change in the journey flow, or the previous version was incorrect?

Reply:
Previous version was incorrect. I've run the smoke and regression tests for standard and clearance journeys and they now pass (were failing before which is why Leszek pushed the ticket back again) :)